diff --git a/cmake/Modules/GrPybind.cmake b/cmake/Modules/GrPybind.cmake
index 7da4e36a56..0518e684a3 100644
--- a/cmake/Modules/GrPybind.cmake
+++ b/cmake/Modules/GrPybind.cmake	
@@ -8,6 +8,9 @@ configure_file(${CMAKE_SOURCE_DIR}/docs/doxygen/pydoc_macros.h ${CMAKE_CURRENT_B
 
 pybind11_add_module(${name}_python ${files})
 
+# Workaround cross compilation issue: https://github.com/gnuradio/gnuradio/issues/5455 <MartB>
+set_target_properties(${name}_python PROPERTIES SUFFIX ".so")
+
 SET(MODULE_NAME ${name})
 if (${name} STREQUAL gr)
     SET(MODULE_NAME "runtime")
@@ -119,6 +122,9 @@ endforeach()
 
 pybind11_add_module(${name}_python ${files})
 
+# Workaround cross compilation issue: https://github.com/gnuradio/gnuradio/issues/5455 <MartB>
+set_target_properties(${name}_python PROPERTIES SUFFIX ".so")
+
 SET(MODULE_NAME ${name})
 if (${name} STREQUAL gr)
     SET(MODULE_NAME "runtime")
@@ -240,6 +246,9 @@ configure_file(${CMAKE_SOURCE_DIR}/docs/doxygen/pydoc_macros.h ${CMAKE_CURRENT_B
 
 pybind11_add_module(${name}_python ${files})
 
+# Workaround cross compilation issue: https://github.com/gnuradio/gnuradio/issues/5455 <MartB>
+set_target_properties(${name}_python PROPERTIES SUFFIX ".so")
+
 SET(MODULE_NAME ${name})
 if (${name} STREQUAL gr)
     SET(MODULE_NAME "runtime")
